lib/sparse/src/index/inverted_index/inverted_index_ram_builder.rs

```diff
@@ -0,0 +1,63 @@
+use common::types::PointOffsetType;
+
+use crate::common::sparse_vector::SparseVector;
+use crate::index::inverted_index::inverted_index_ram::InvertedIndexRam;
+use crate::index::posting_list::PostingBuilder;
+
+/// Builder for InvertedIndexRam
+pub struct InvertedIndexBuilder {
+    pub posting_builders: Vec<PostingBuilder>,
+    pub vector_count: usize,
+}
+
+impl Default for InvertedIndexBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl InvertedIndexBuilder {
+    pub fn new() -> InvertedIndexBuilder {
+        InvertedIndexBuilder {
+            posting_builders: Vec::new(),
+            vector_count: 0,
+        }
+    }
+
+    /// Add a vector to the inverted index builder
+    pub fn add(&mut self, id: PointOffsetType, vector: SparseVector) {
+        for (dim_id, weight) in vector.indices.into_iter().zip(vector.values.into_iter()) {
+            let dim_id = dim_id as usize;
+            match self.posting_builders.get_mut(dim_id) {
```

Comment:
:+1:

Reply:
Thank you, that's a very clean way of doing it :+1:

Reply:
Looking at this in more details, it sadly does not work.

Because `If new_len is less than len, the Vec is simply truncated.`.

I would need an operation that only expend if necessary to follow this scheme.